### PR TITLE
[DNS] Use client from ctx in `zone_v2` and `recordset_v2`

### PR DESCRIPTION
**Pull request by anton-sidelnikov**
## PR Checklist

* [x] Refers to: #1848
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.
* [x] Release notes added.

## Acceptance Steps Performed

```
=== RUN   TestAccDNSV2Zone_basic
--- PASS: TestAccDNSV2Zone_basic (25.17s)
=== RUN   TestAccDNSV2Zone_unDotted
--- PASS: TestAccDNSV2Zone_unDotted (15.62s)
=== RUN   TestAccDNSV2Zone_private
--- PASS: TestAccDNSV2Zone_private (16.48s)
=== RUN   TestAccDNSV2Zone_readTTL
--- PASS: TestAccDNSV2Zone_readTTL (15.39s)
=== RUN   TestAccDNSV2Zone_timeout
--- PASS: TestAccDNSV2Zone_timeout (15.52s)
=== RUN   TestAccCheckDNSV2Zone_routerValidation
--- PASS: TestAccCheckDNSV2Zone_routerValidation (3.20s)
PASS

Process finished with the exit code 0

=== RUN   TestAccDNSV2RecordSet_basic
--- PASS: TestAccDNSV2RecordSet_basic (41.05s)
=== RUN   TestAccDNSV2RecordSet_unDotted
--- PASS: TestAccDNSV2RecordSet_unDotted (25.87s)
=== RUN   TestAccDNSV2RecordSet_childFirst
--- PASS: TestAccDNSV2RecordSet_childFirst (35.73s)
=== RUN   TestAccDNSV2RecordSet_readTTL
--- PASS: TestAccDNSV2RecordSet_readTTL (26.11s)
=== RUN   TestAccDNSV2RecordSet_timeout
--- PASS: TestAccDNSV2RecordSet_timeout (25.84s)
=== RUN   TestAccDNSV2RecordSet_shared
--- PASS: TestAccDNSV2RecordSet_shared (35.22s)
=== RUN   TestAccDNSV2RecordSet_txt
--- PASS: TestAccDNSV2RecordSet_txt (35.68s)
PASS

Process finished with the exit code 0

```

### CODE REVIEW

File: opentelekomcloud/services/dns/resource_opentelekomcloud_dns_recordset_v2.go [hunk 1]
```diff
@@ -150,19 +152,20 @@ func resourceDNSRecordSetV2Create(ctx context.Context, d *schema.ResourceData, m
...
-		Timeout:    d.Timeout(schema.TimeoutCreate),
-		Delay:      5 * time.Second,
-		MinTimeout: 3 * time.Second,
+		Target:       []string{"ACTIVE"},
+		Pending:      []string{"PENDING"},
+		Refresh:      waitForDNSRecordSet(client, zoneID, recordSet.ID),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        5 * time.Second,
+		MinTimeout:   3 * time.Second,
+		PollInterval: 2,
```
Comment (lego963) [inline opentelekomcloud/services/dns/resource_opentelekomcloud_dns_recordset_v2.go]: 2*time.Second

File: opentelekomcloud/services/dns/resource_opentelekomcloud_dns_recordset_v2.go [hunk 2]
```diff
@@ -294,19 +302,20 @@ func resourceDNSRecordSetV2Update(ctx context.Context, d *schema.ResourceData, m
...
-		Timeout:    d.Timeout(schema.TimeoutUpdate),
-		Delay:      5 * time.Second,
-		MinTimeout: 3 * time.Second,
+		Target:       []string{"ACTIVE"},
+		Pending:      []string{"PENDING"},
+		Refresh:      waitForDNSRecordSet(client, zoneID, recordsetID),
+		Timeout:      d.Timeout(schema.TimeoutUpdate),
+		Delay:        5 * time.Second,
+		MinTimeout:   3 * time.Second,
+		PollInterval: 2,
```
Comment (lego963) [inline opentelekomcloud/services/dns/resource_opentelekomcloud_dns_recordset_v2.go]: and here

File: opentelekomcloud/services/dns/resource_opentelekomcloud_dns_recordset_v2.go [hunk 3]
```diff
@@ -350,19 +362,20 @@ func resourceDNSRecordSetV2Delete(ctx context.Context, d *schema.ResourceData, m
...
-		Timeout:    d.Timeout(schema.TimeoutDelete),
-		Delay:      5 * time.Second,
-		MinTimeout: 3 * time.Second,
+		Target:       []string{"DELETED"},
+		Pending:      []string{"ACTIVE", "PENDING", "ERROR"},
+		Refresh:      waitForDNSRecordSet(client, zoneID, recordsetID),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        5 * time.Second,
+		MinTimeout:   3 * time.Second,
+		PollInterval: 2,
```
Comment (lego963) [inline opentelekomcloud/services/dns/resource_opentelekomcloud_dns_recordset_v2.go]: +

**Review by lego963 — CHANGES_REQUESTED** [review]
add `*time.Second` in PollInterval pls

**Review by artem-lifshits — APPROVED** [review]

**Review by lego963 — APPROVED** [review]

**Review by vladimirvshivkov — APPROVED** [review]

**Pull request merged**